api/api.go

```diff
@@ -22,6 +22,16 @@ import (
 	"time"
 )
 
+// ProtocolHandler handles incoming message according to the implemented protocol.
+//
+// Handle verifies and processes incoming messages according to the protocol.
+// If there is any message produced in reply, it will be send to reply
+// channel, otherwise nil channel is returned. The return value new
+// indicates that the message has not been processed before.
+type ProtocolHandler interface {
```

Comment:
This package was supposed to define the API between MinBFT core and external components, but not components of the core. If we decide to expose those components, it would be better to move those to separate packages, e.g. under `core`. Then those packages would define their API.

Reply:
I am not sure I understand the idea. Wouldn't be enough to just move `ProtocolHandler` to `core`?

Reply:
Either to `core` or maybe better to a new sub-package under `core`. That sub-package might encapsulate all related code, then.

Reply:
Wouldn't it make sense to put `ProtocolHandler` under `core` and move the code specific to minbft to either `core/minbft` or `core/default`?